src/ol_infrastructure/substructure/vault/secrets/__main__.py

```diff
@@ -36,6 +38,68 @@
         data_json=json.dumps(data),
     )
 
+# Configure secret-dev mount and keycloak auth
+if "QA" in stack_info.name:
+    # Create the secret mount used for storing env secrets for developers
+    dev_vault_mount = vault.Mount(
+        f"ol-dev-configuration-secrets-mount-{stack_info.env_suffix}",
+        path="secret-dev",
+        type="kv-v2",
+        options={"version": 2},
+        description="Storage of configuration secrets used by Devs",
+        opts=ResourceOptions(delete_before_replace=True),
+    )
+
+    # Read MIT Open vault secrets
+    mitopen_vault_secrets = read_yaml_secrets(
+        Path(f"mitopen/secrets.{stack_info.env_suffix}.yaml"),
+    )
+
+    vault.generic.Secret(
+        f"ol-dev-configuration-secrets-{stack_info.env_suffix}",
+        path=dev_vault_mount.path.apply("{}/mitopen/secrets".format),
+        data_json=json.dumps(mitopen_vault_secrets),
+    )
+
+    # Enable OIDC auth method and configure it with Keycloak
+    vault_oidc_keycloak_auth = vault.jwt.AuthBackend(
+        "vault-oidc-keycloak-backend",
+        path="oidc",
+        type="oidc",
+        description="OIDC auth Keycloak integration for use with dev vault client cli",
+        oidc_discovery_url=f"{keycloak_config.get("url")}/realms/ol-platform-engineering",
+        oidc_client_id=keycloak_config.get("client_id"),
+        oidc_client_secret=keycloak_config.get("client_secret"),
+        default_role="local-developer",
+        opts=ResourceOptions(delete_before_replace=True),
+    )
+
+    # Local Developer policy definition
+    local_developer_policy = vault.Policy(
+        "local-developer-policy",
+        name="local-developer",
+        policy=(Path(__file__).resolve())
+        .parent.parent.joinpath("policies/developer/local_developer_policy.hcl")
+        .read_text(),
+    )
+
+    # Configure OIDC role
+    local_dev_role = vault.jwt.AuthBackendRole(
+        "local-dev-role",
+        backend=vault_oidc_keycloak_auth.path,
+        role_name="local-dev",
+        token_policies=[
+            local_developer_policy.name,
+        ],
+        allowed_redirect_uris=[
+            "http://localhost:8250/oidc/callback",
+            f"{vault_config.get('address')}/ui/vault/auth/oidc/oidc/callback",
+        ],
+        bound_audiences=[f"{keycloak_config.get('client_id')}"],
```

Comment:
```suggestion
        bound_audiences=[keycloak_config.get('client_id')],
```
Is there a reason that it needs to be templated as an `f` string?